### PR TITLE
WT-7548 Create macro to identify dhandles that behave as a Btree

### PR DESCRIPTION
**Pull request by ddanderson**

### CODE REVIEW

**Comment by ddanderson** [comment]
@sueloverso and @keitharnoldsmith, this is ready for review.  Please see my comment in https://jira.mongodb.org/browse/WT-7548 as to why I kept the tiered dhandle out of the macro def (for now) and did not introduce a macro for URIs.

File: src/cursor/cur_file.c [hunk 1]
```diff
@@ -821,7 +821,7 @@ __wt_curfile_open(WT_SESSION_IMPL *session, const char *uri, WT_CURSOR *owner, c
     if (bulk)
         LF_SET(WT_BTREE_BULK | WT_DHANDLE_EXCLUSIVE);
 
-    WT_ASSERT(session, WT_PREFIX_MATCH(uri, "file:"));
+    WT_ASSERT(session, WT_PREFIX_MATCH(uri, "file:") || WT_PREFIX_MATCH(uri, "tiered:"));
```
Comment (keitharnoldsmith) [inline src/cursor/cur_file.c:824]: This change is also in my PR for WT-7440.  So we'll have a minor collision here.

**Review by keitharnoldsmith — APPROVED** [review]
LGTM

**Review by sueloverso — APPROVED** [review]
LGTM

**Pull request merged**